src/features/game/expansion/components/resources/components/TreeReward.tsx

```diff
@@ -0,0 +1,127 @@
+import { Panel } from "components/ui/Panel";
+import React, { useContext, useEffect, useRef, useState } from "react";
+import { Modal } from "react-bootstrap";
+
+import { Reward } from "features/game/types/game";
+
+import { Button } from "components/ui/Button";
+import { ITEM_DETAILS } from "features/game/types/images";
+import { Context } from "features/game/GameProvider";
+
+import token from "assets/icons/token_2.png";
+import { StopTheGoblins } from "features/farming/crops/components/StopTheGoblins";
+import { ChestCaptcha } from "features/farming/crops/components/ChestCaptcha";
+
+interface Props {
+  reward: Reward | null;
+  treeIndex: number;
+  expansionIndex: number;
+  onCollected: (success: boolean) => void;
+}
+
+type Challenge = "goblins" | "chest";
+
+export const TreeReward: React.FC<Props> = ({
+  reward,
+  onCollected,
+  expansionIndex,
+  treeIndex,
+}) => {
+  const { gameService } = useContext(Context);
+  const [opened, setOpened] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const rewardType = "Tree";
+  const challenge = useRef<Challenge>(
+    Math.random() > 0.3 ? "chest" : "goblins"
+  );
+
+  useEffect(() => {
+    if (reward) {
+      setLoading(true);
+      setTimeout(() => setLoading(false), 750);
+    }
+  }, [reward]);
+
+  if (!reward) {
+    return null;
+  }
+
+  const open = () => {
+    setOpened(true);
+    gameService.send("treeReward.collected", {
```

Comment:
Since the only thing that is different there is this action I wonder if we could pass the action as a prop and just use the one `Reward` component in Land Expansion?

Reply:
> Since the only thing that is different there is this action I wonder if we could pass the action as a prop and just use the one `Reward` component in Land Expansion?

Dont this create too much nested conditions to check the action and the different rewards?